src/apify/log.py

```diff
@@ -0,0 +1,85 @@
+import json
+import logging
+import textwrap
+import traceback
+from typing import Any, Dict
+
+from colorama import Fore, Style, just_fix_windows_console
+
+just_fix_windows_console()
+
+
+# Name of the logger used throughout the library (resolves to 'apify')
+logger_name = __name__.split('.')[0]
+
+# Logger used throughout the library
+logger = logging.getLogger(logger_name)
+
+
+_LOG_LEVEL_COLOR = {
+    logging.DEBUG: Fore.BLUE,
+    logging.INFO: Fore.GREEN,
+    logging.WARNING: Fore.YELLOW,
+    logging.ERROR: Fore.RED,
+    logging.CRITICAL: Fore.RED,
+}
+
+_LOG_LEVEL_SHORT_ALIAS = {
+    logging.DEBUG: 'DEBUG',
+    logging.INFO: 'INFO ',
+    logging.WARNING: 'WARN ',
+    logging.ERROR: 'ERROR',
+}
+
+# So that all the log messages have the same alignment
+_LOG_MESSAGE_INDENT = ' ' * 6
+
+
+class ActorLogFormatter(logging.Formatter):
+    """Log formatter that prints out the log message nicely formatted, with colored level and stringified extra fields."""
+
+    empty_record = logging.LogRecord('dummy', 0, 'dummy', 0, 'dummy', None, None)
```

Comment:
wtf is this

Reply:
Ah yeah this deserves a comment, will add it 😄